WIP: Added scripts that produce the results for the site visit and cleanup

freewym opened the pull request:
…nd do 2nd-pass decoding

freewym commented on egs/material/s5/local/chain/decode_test.sh:
```diff
@@ -97,10 +113,66 @@ done
+wait
+[ -f $dir/.error ] && echo "$0: there was a problem while decoding" && exit 1
+
+if [ $stage -le 19 ]; then
```
@danpovey I extracted ivectors based on the uniform segments and vad weights in this stage and decode in the next stage. But WER was not reduced. Am I doing things right?

danpovey commented:
Thanks!  Can you please say what the WERs were on the two datasets, and what the corresponding baselines were?  We'll need this for the reports.

danpovey commented:
.. and Yiming, let us know if the ins/del/sub is significantly different.
And @vimalmanohar, could you please check this PR to see if it makes sense?

freewym commented:
first-pass decoding results on dev set:
%WER 39.28 [ 24408 / 62144, 3470 ins, 5731 del, 15207 sub ] exp/chain/tdnn_lstm1a_sp_ld5/decode_dev/wer_10_0.0

first-pass decoding results on the test set "analysis1-segmented":
%WER 66.13 [ 44626 / 67484, 1675 ins, 19158 del, 23793 sub ]

2nd-pass results (after recomputing ivectors) on the test set "analysis1-segmented":
%WER 66.24 [ 44700 / 67484, 1698 ins, 19343 del, 23659 sub ]

Do I need to give 2nd pass decoding result on dev set?

danpovey commented:
So what segment boundaries (i.e. what 'segments' file) were you using in
the 2nd-pass results?  Did that change?

I think this only makes sense on the analysis dataset, because it's
unsegmented, but please try it for both languages.  For these ones, it's
not clear what the language is.


On Sun, Feb 11, 2018 at 7:46 PM, Yiming Wang <notifications@github.com>
wrote:

> first-pass decoding results on dev set:
> %WER 39.28 [ 24408 / 62144, 3470 ins, 5731 del, 15207 sub ]
> exp/chain/tdnn_lstm1a_sp_ld5/decode_dev/wer_10_0.0
>
> first-pass decoding results on the test set "analysis1-segmented":
> %WER 66.13 [ 44626 / 67484, 1675 ins, 19158 del, 23793 sub ]
>
> 2nd-pass results (after recomputing ivectors) on the test set
> "analysis1-segmented":
> %WER 66.24 [ 44700 / 67484, 1698 ins, 19343 del, 23659 sub ]
>
> Do I need to give 2nd pass decoding result on dev set?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mahsa7823/kaldi/pull/1#issuecomment-364806887>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVuwJcegrHuIhmTJaID9E5JOifvO0yks5tT4nxgaJpZM4SBgmI>
> .
>

freewym commented:
It was the same as the 1st passing decoding, i.e. data/analysis1/uniform_sub_segments.

The above results is in language swahili, will try on the other one.

danpovey commented:
OK.
Try creating a data directory with a segments file determined by the ctm
output of the 1st pass decoding (you'll have to make a program to create
the segments file, or ask Vimal if one exists), and in using that
directory, create ivectors using the normal script and then decode.
Dan


On Sun, Feb 11, 2018 at 8:11 PM, Yiming Wang <notifications@github.com>
wrote:

> It was the same as the 1st passing decoding, i.e.
> data/analysis1/uniform_sub_segments.
>
> The above results is in language swahili, will try on the other one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mahsa7823/kaldi/pull/1#issuecomment-364809047>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu2V9YwjllnPvYXZ72ZDBG1HASQIJks5tT4_DgaJpZM4SBgmI>
> .
>

danpovey commented:
Bear in mind that we'll have to evaluate how much improvement this gives before committing it.

danpovey commented:
... and Yiming, see if you can resolve the conflicts.

freewym commented:
I just want to show what I have changed: I modified the script as in my last commit to resegment the test data. For swahili, WER is reduced from 66.13 to 65.69, which is not significant. Could @danpovey or @vimalmanohar check if I did it correctly?

danpovey commented:
OK, cool.  @vimalmanohar can you please check?

freewym commented:
... the main changes are in stage 19.

vimalmanohar commented:
It looks ok. You should probably try offline i-vectors and check the
results. If it still does not work, train a proper SAD system like in
https://github.com/kaldi-asr/kaldi/blob/master/egs/swbd/s5c/local/run_asr_segmentation.sh

On Thu, Feb 15, 2018 at 5:24 PM Yiming Wang <notifications@github.com>
wrote:

> ... the main changes are in stage 19.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mahsa7823/kaldi/pull/1#issuecomment-366082028>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEATV6Xvy6fa7k_vglmemjVRfLAiUA63ks5tVK6WgaJpZM4SBgmI>
> .
>
-- 
Vimal Manohar
PhD Student
Electrical & Computer Engineering
Johns Hopkins University

danpovey commented:
It looks plausible to me.   Can you also report results on the other
language?


On Thu, Feb 15, 2018 at 5:24 PM, Yiming Wang <notifications@github.com>
wrote:

> ... the main changes are in stage 19.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mahsa7823/kaldi/pull/1#issuecomment-366082028>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu_STfFMpzczpzRCFlYf4WItua0sQks5tVK6VgaJpZM4SBgmI>
> .
>

vimalmanohar commented on egs/material/s5/local/chain/decode_test.sh:
```diff
@@ -110,13 +128,104 @@ for data in $test_sets; do
+    # get alignment from lattice
+    nj=`cat ${dir}/decode_${data}/num_jobs` || exit 1;
+    $cmd JOB=1:$nj ${dir}/decode_${data}/log/generate_alignments.JOB.log \
+    lattice-best-path --acoustic-scale=0.3 \
```
acwt should be 0.1 usually.

freewym commented:
@mahsa7823 I have done all the cleanups. Please take a look and let me know if there are any problems.

freewym commented:
@danpovey  I didn't include the tdnn-F recipe since the relavant binary code seems not have been merged to master and there are some problem when trying to read models with tdnn-F structures.

danpovey commented:
It has been merged to master.  probably you didn't merge this branch with master.  But I can merge anyway, no need to wait on that.

danpovey commented:
actually let's leave this as a separate branch for now, I don't see a need to merge it at this point.
But keep it up-to-date by merging from master.

The TDNN-F thing that you are talking about-- yes, it does require some dropout code that's new, from a branch I haven't merged yet.  I'll get back to you about that later.

danpovey commented on egs/material/s5/local/chain/decode_test.sh:
```diff
@@ -1,44 +1,41 @@
-
-tlstm_affix=1a   # affix for the TDNN-LSTM directory name
-tree_affix=
+dir=exp/chain/tdnn_lstm1a_sp_ld5
```
please don't include the _ld5 suffix, here or where you build the model.

danpovey commented on scripts/rnnlm/lmrescore_pruned.sh:
```diff
@@ -17,6 +17,7 @@ max_ngram_order=4 # Approximate the lattice-rescoring by limiting the max-ngram-
                   # exploding exponentially. Details of the n-gram approximation
                   # method are described in section 2.3 of the paper
                   # http://www.danielpovey.com/files/2018_icassp_lattice_pruning.pdm
+max_arcs=         # limit the max arcs in lattice while rescoring
```
can you please make a separate PR for this?  or have Ke or Hainan do it?

freewym replied:
Sure. A PR to master or material_basic branch?

danpovey commented:
to master-- just with the scripts/rnnlm stuff.


On Wed, Feb 28, 2018 at 7:40 PM, Yiming Wang <notifications@github.com>
wrote:

> *@freewym* commented on this pull request.
> ------------------------------
>
> In scripts/rnnlm/lmrescore_pruned.sh
> <https://github.com/mahsa7823/kaldi/pull/1#discussion_r171432977>:
>
> > @@ -17,6 +17,7 @@ max_ngram_order=4 # Approximate the lattice-rescoring by limiting the max-ngram-
>                    # exploding exponentially. Details of the n-gram approximation
>                    # method are described in section 2.3 of the paper
>                    # http://www.danielpovey.com/files/2018_icassp_lattice_pruning.pdm
> +max_arcs=         # limit the max arcs in lattice while rescoring
>
> Sure. A PR to master or material_basic branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mahsa7823/kaldi/pull/1#discussion_r171432977>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu0O4hDkpfJfIgPU2Uobo99EYt-rOks5tZ0N7gaJpZM4SBgmI>
> .
>

freewym commented:
@mahsa7823 Probably you need to do "git pull --rebase origin master" again to resolve the conflicts, since I just did it for my branch.

mahsa7823 commented on egs/material/s5/local/postprocess_test.sh:
```diff
@@ -4,35 +4,34 @@ set -e -o pipefail
-    $dir/decode_$data/
+steps/get_ctm_fast.sh --frame-shift 0.03 \
+  data/${data}_hires/ ${graph_dir} \
+  ${decode_dir} ${decode_dir}/score_10/
```
Dan previously said that the directory structure was rather strange: "score_${lmwt} implies that it's part of some kind of scoring script (which it isn't).  Is there a good reason for this?  I think it should probably be changed."

freewym replied:
I see from swbd that score_$lmwt are also directly in $decode_dir

mahsa7823 commented on egs/material/s5/run.sh:
```diff
@@ -153,3 +156,106 @@ if [ $stage -le 11 ]; then
       exp/tri3/graph data/$test exp/tri3/decode_$test
   done
 fi
+
```
It would be better to move this part to a separate "preprocess" script.
